server/src/main/java/io/druid/guice/StorageNodeModule.java

```diff
@@ -65,6 +65,7 @@ public DruidServerMetadata getMetadata(@Self DruidNode node, @Nullable NodeTypeC
     return new DruidServerMetadata(
         node.getHostAndPort(),
         node.getHostAndPort(),
+        node.getHostAndTlsPort(),
```

Comment:
Let me fix this and make sure same problem is not happening elsewhere where `getHostAndPort` is used.